packages/transport-tcp/test/socket-to-conn.spec.ts

```diff
@@ -287,6 +289,52 @@ describe('socket-to-conn', () => {
     expect(serverSocket.destroyed).to.be.true()
   })
 
+  it('should not close MultiaddrConnection twice', async () => {
+    const loggerStub = Sinon.stub()
+    const logger: ComponentLogger = {
+      forComponent: () => loggerStub as unknown as Logger
+    };
+    ({ server, clientSocket, serverSocket } = await setup())
+
+    // promise that is resolved when our outgoing socket is closed
+    const serverClosed = defer<boolean>()
+
+    const inboundMaConn = toMultiaddrConnection(serverSocket, {
+      socketInactivityTimeout: 100,
+      socketCloseTimeout: 10,
+      logger
+    })
+    expect(inboundMaConn.timeline.open).to.be.ok()
+    expect(inboundMaConn.timeline.close).to.not.be.ok()
+
+    clientSocket.once('error', () => {})
+
+    serverSocket.once('close', () => {
+      serverClosed.resolve(true)
+    })
+
+    // send some data between the client and server
+    clientSocket.write('hello')
+    serverSocket.write('goodbye')
+
+    // the 2nd call should return immediately
+    await Promise.all([
+      inboundMaConn.close(),
+      inboundMaConn.close()
+    ])
+
+    expect(loggerStub.calledWithMatch('socket is either closed, closing, or already destroyed')).to.be.true()
+
+    // server socket was closed for reading and writing
+    await expect(serverClosed.promise).to.eventually.be.true()
+
+    // the connection closing was recorded
+    expect(inboundMaConn.timeline.close).to.be.a('number')
+
+    // server socket is destroyed
+    expect(serverSocket.destroyed).to.be.true()
```

Comment:
Assert that we only close the socket once:

```suggestion
    expect(serverSocket.destroyed).to.be.true()

    // the server socket was only closed once
    expect(serverSocketDestroySpy.callCount).to.equal(1)
```